src/operations/command.ts

```diff
@@ -75,21 +100,45 @@ export abstract class CommandOperation<
     const propertyProvider = this.hasAspect(Aspect.NO_INHERIT_OPTIONS) ? undefined : parent;
     this.readPreference = this.hasAspect(Aspect.WRITE_OPERATION)
       ? ReadPreference.primary
-      : ReadPreference.resolve(propertyProvider, this.options);
-    this.readConcern = resolveReadConcern(propertyProvider, this.options);
-    this.writeConcern = resolveWriteConcern(propertyProvider, this.options);
+      : ReadPreference.resolve(propertyProvider, options);
+    this.readConcern = resolveReadConcern(propertyProvider, options);
+    this.writeConcern = resolveWriteConcern(propertyProvider, options);
     this.explain = false;
     this.fullResponse =
       options && typeof options.fullResponse === 'boolean' ? options.fullResponse : false;
 
+    // if (this.writeConcern && this.writeConcern.w === 0) {
+    //   if (this.session && this.session.explicit) {
+    //     throw new MongoError('Cannot have explicit session with unacknowledged writes');
+    //   }
+    //   return;
+    // }
```

Comment:
I ran into a test failure when trying to add a few more operations to this PR that relates to this in `applySession` which is commented saying it should occur at command construction, I'll still have to discover the bug, but shall we move this here?
Maybe as part of removing legacy operations so everything would go through this path.